treeherder/config/settings.py

```diff
@@ -196,6 +218,58 @@
     }
 }
 
+# HTTPS
+USE_X_FORWARDED_HOST = True
+USE_X_FORWARDED_PORT = True
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+if SITE_URL.startswith('https://'):
+    SECURE_SSL_REDIRECT = True
+    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
+    SECURE_HSTS_PRELOAD = True
+    SECURE_HSTS_SECONDS = int(timedelta(days=365).total_seconds())
+    # Mark session and CSRF cookies as being HTTPS-only.
+    CSRF_COOKIE_SECURE = True
+    SESSION_COOKIE_SECURE = True
+
+SECURE_CONTENT_TYPE_NOSNIFF = True  # Set the `X-Content-Type-Options` header to `nosniff`.
+SECURE_BROWSER_XSS_FILTER = True  # Sets the `X-XSS-Protection` header.
+
+# System Checks
+SILENCED_SYSTEM_CHECKS = [
+    # We can't set CSRF_COOKIE_HTTPONLY to True since the requests to the API
+    # made using Angular's `httpProvider` require access to the cookie.
+    'security.W017',
+    # We can't set X_FRAME_OPTIONS to DENY since renewal of auth token requires
+    # opening an invisible iframe with the same origin.
+    'security.W019'
+]
+
+# User Agents
+# User agents which will be blocked from making requests to the site.
+DISALLOWED_USER_AGENTS = (
+    # Note: This intentionally does not match the command line curl
+    # tool's default User Agent, only the library used by eg PHP.
+    re.compile(r'^libcurl/'),
+    re.compile(r'^Python-urllib/'),
+    re.compile(r'^python-requests/'),
+    # ActiveData blocked for excessive API requests (bug 1289830)
+    re.compile(r'^ActiveData-ETL'),
+)
+
+
+# THIRD PARTY APPS
+
+# Auth0 setup
+AUTH0_DOMAIN = env('AUTH0_DOMAIN', default="auth.mozilla.auth0.com")
+AUTH0_CLIENTID = env('AUTH0_CLIENTID', default="q8fZZFfGEmSB2c5uSI8hOkKdDGXnlo5z")
+
+# Set the `X-Frame-Options` header, which forbids embedding of site pages in frames other than origin.
+# AUTH0 renewal opens the auth handler page in an invisible frame
+# hence requiring the need to support frames with same origin
+X_FRAME_OPTIONS = 'SAMEORIGIN'
```

Comment:
I think I mistakenly left this here because of the context of the comment.  However given it's currently set to the default value, should we remove it?

Reply:
Yeah lets remove it (it exists in this file since it was previoulsy set to `DENY` but had to be adjusted when we switched to Auth0) - but I'd perhaps move some of the additional comment detail into the existing `DISALLOWED_USER_AGENTS` comment.

Reply:
Done in e714d7056